common/lib/common-utils/src/bufferBrowser.ts

```diff
@@ -66,6 +71,8 @@ export class IsoBuffer extends Uint8Array {
     static from(value, encodingOrOffset?, length?): IsoBuffer {
         if (typeof value === "string") {
             return IsoBuffer.fromString(value, encodingOrOffset as string | undefined);
+        } else if (value instanceof IsoBuffer) {
```

Comment:
Ok I think I see the problem now.  We return `ArrayBufferLike` in those other methods but we only check for `instanceof ArrayBuffer` here.  The Node `Buffer` class accepts an `ArrayBuffer | SharedArrayBuffer` which appears equivalent to `ArrayBufferLike`.  So I think changing the typing here to `ArrayBufferLike` instead of just `ArrayBuffer` is more equivalent (though your change may be faster?)

Reply:
I'm pretty sure you can't do instanceof ArrayBufferLike check as there is no such function. instanceof checks if function (constructor of a class) exists in proto chain of object you are checking. But it's not a class - it does not exists at runtime at all (as it's TS construct - it compiles to nothing in JS)

type ArrayBufferLike = ArrayBufferTypes[keyof ArrayBufferTypes];

IMHO, the core of the problem is that IsoBuffer is exposed - I think it should be internal implementation detail that is never exposed not exposed and instead only helper methods like buggerToString exposed.